alacritty/src/input/keyboard.rs

```diff
@@ -589,55 +588,54 @@ impl SequenceBuilder {
             return None;
         }
 
-        let named = match key.logical_key {
-            Key::Named(named) => named,
+        let mut base = match key.logical_key.as_ref() {
+            Key::Named(NamedKey::Tab) => "9",
+            Key::Named(NamedKey::Enter) => "13",
+            Key::Named(NamedKey::Escape) => "27",
+            Key::Character(" ") => "32",
+            Key::Named(NamedKey::Backspace) => "127",
+            Key::Named(_) => "",
             _ => return None,
         };
 
-        let base = match named {
-            NamedKey::Tab => "9",
-            NamedKey::Enter => "13",
-            NamedKey::Escape => "27",
-            NamedKey::Space => "32",
-            NamedKey::Backspace => "127",
-            _ => "",
-        };
-
         // Fail when the key is not a named control character and the active mode prohibits us
         // from encoding modifier keys.
         if !self.kitty_encode_all && base.is_empty() {
             return None;
         }
 
-        let base = match (named, key.location) {
-            (NamedKey::Shift, KeyLocation::Left) => "57441",
-            (NamedKey::Control, KeyLocation::Left) => "57442",
-            (NamedKey::Alt, KeyLocation::Left) => "57443",
-            (NamedKey::Super, KeyLocation::Left) => "57444",
-            (NamedKey::Hyper, KeyLocation::Left) => "57445",
-            (NamedKey::Meta, KeyLocation::Left) => "57446",
-            (NamedKey::Shift, _) => "57447",
-            (NamedKey::Control, _) => "57448",
-            (NamedKey::Alt, _) => "57449",
-            (NamedKey::Super, _) => "57450",
-            (NamedKey::Hyper, _) => "57451",
-            (NamedKey::Meta, _) => "57452",
-            (NamedKey::CapsLock, _) => "57358",
-            (NamedKey::NumLock, _) => "57360",
-            _ => base,
-        };
-
         // NOTE: Kitty's protocol mandates that the modifier state is applied before
         // key press, however winit sends them after the key press, so for modifiers
         // itself apply the state based on keysyms and not the _actual_ modifiers
         // state, which is how kitty is doing so and what is suggested in such case.
         let press = key.state.is_pressed();
-        match named {
-            NamedKey::Shift => mods.set(SequenceModifiers::SHIFT, press),
-            NamedKey::Control => mods.set(SequenceModifiers::CONTROL, press),
-            NamedKey::Alt => mods.set(SequenceModifiers::ALT, press),
-            NamedKey::Super => mods.set(SequenceModifiers::SUPER, press),
-            _ => (),
+
+        if let Key::Named(named) = key.logical_key.as_ref() {
+            base = match (named, key.location) {
+                (NamedKey::Shift, KeyLocation::Left) => "57441",
+                (NamedKey::Control, KeyLocation::Left) => "57442",
+                (NamedKey::Alt, KeyLocation::Left) => "57443",
+                (NamedKey::Meta, KeyLocation::Left) => "57444",
+                #[allow(deprecated)]
```

Comment:
What's deprecated here? `NamedKey::Hyper`? If so, what's the alternative going to be?

Reply:
> there's no keyboard with them

You're not using a [Space-cadet](https://en.wikipedia.org/wiki/Space-cadet_keyboard) keyboard?